File: Lista4_ED1e/ex21.c
#include<stdio.h>
//21 - Fatorial

    int main(){
        int i, j, total=1;

        for(i=1; i<=10; i++){
            for(j=i; j>=1; j--){
                total*=j;
            }
            printf("%d! = %d\n", i, total);
            total=1;
        }


    }

File: ProLog/Roteiro 3/ex21.pl
fatorial(0,1). 
fatorial(M,N):-
   M>0,
   M1 is M-1,
   fatorial(M1,N1),
   N is M*N1.

ncombinacoes(M, P, N):-
    fatorial(M, X),
    fatorial(P, Y),
    Z is M-P,
    fatorial(Z, W),
    N is X/(Y*W).